### PR TITLE
Update README to add instruction for building C++

### PR DESCRIPTION
**Pull request by eddyxu**
Added instructions to build C++ on `macOS` and `Ubuntu 22.04`

### CODE REVIEW

File: cpp/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# On macOS 12+
+brew install apache-arrow cmake protobuf
+# Optionally, build document
+brew install doxygen
+```
+
+```sh
+# On Ubuntu 22.04
+# Follow https://arrow.apache.org/install/
+sudo apt update
```
Comment (changhiskhan) [inline cpp/README.md:38]: I would say just link to arrow is fine. If they update their instructions there's no need to have to update ours too

File: cpp/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+sudo apt install -y -V ca-certificates lsb-release wget
+wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+sudo apt install -y -V ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+sudo apt update
+
+# Install arrow, protobuf, cmake
+sudo apt install -y -V libarrow-dev libarrow-dataset-dev libparquet-dev libarrow-python-dev cmake libprotobuf-dev protobuf-compiler
```
Comment (changhiskhan) [inline cpp/README.md:45]: Do we need to install protobuf-compiler explicitly? I forgot if it's a libprotobuf-dev dep

File: cpp/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Lance: A Columnar File Format
```
Comment (changhiskhan) [inline cpp/README.md:1]: Top level says data format and this says file format

**Review by changhiskhan — APPROVED** [review]

File: cpp/README.md (hunk 1 above)
Comment (eddyxu) [reply cpp/README.md:38]: done

File: cpp/README.md (hunk 2 above)
Comment (eddyxu) [reply cpp/README.md:45]: Good call. lemme check

File: cpp/README.md (hunk 3 above)
Comment (eddyxu) [reply cpp/README.md:1]: Fixed. To be all data format.

**Pull request merged**